chromium_src/chrome/browser/first_run/first_run.cc

```diff
@@ -1,19 +1,61 @@
-/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#define IsMetricsReportingOptIn IsMetricsReportingOptIn_UnUsed
+#include "chrome/browser/first_run/first_run.h"
+
+#include "base/command_line.h"
+#include "base/notreached.h"
+#include "brave/browser/metrics/switches.h"
+#include "build/build_config.h"
+#include "chrome/common/channel_info.h"
+#include "components/version_info/channel.h"
+
+#define IsMetricsReportingOptIn IsMetricsReportingOptIn_ChromiumImpl
 #include "src/chrome/browser/first_run/first_run.cc"
 #undef IsMetricsReportingOptIn
 
 namespace first_run {
 
-// This is only used for determine whether crash report checkbox in the first
-// run dialog is checked or not by default. See the comments of this upstream
-// function. Returning true means crash report is unchecked by default.
+bool IsMetricsReportingOptIn(version_info::Channel channel) {
+  switch (channel) {
+    case version_info::Channel::STABLE:
+#if BUILDFLAG(IS_ANDROID)
+      // Change both to false when android implements ask on first crash
+      // dialog
+      return false;
+#else
+      return true;
+#endif
+    case version_info::Channel::BETA:  // fall through
+    case version_info::Channel::DEV:   // fall through
```

Comment:
Could this be better?

```suggestion
    case version_info::Channel::BETA:  [[fallthrough]]
    case version_info::Channel::DEV:   [[fallthrough]]
```

Reply:
this code was just copied from somewhere else, but I can update it